s3watcher/lambda/lib/Lambda.ts

```diff
@@ -0,0 +1,91 @@
+import { S3EventRecord } from "aws-lambda"
+import { S3 } from "aws-sdk"
+
+export interface ImportAction {
+  bucket: string
+  path: string[]
+  key: string
+  filename: string
+  size: number
+}
+
+export interface IngestConfig {
+  region: string
+  baseUrl: string
+  apiKey: string
+  failBucket: string
+  s3UrlExpiry: number
+  stage: string
+}
+
+export const createActionFromNotification = function (
+  record: S3EventRecord
+): ImportAction {
+  const e = record.s3
+
+  const normaliseKey = function (key: string) {
+    return decodeURIComponent(key.replace(/\+/g, " "))
+  }
+  const key = normaliseKey(e.object.key)
+  const bucket = e.bucket.name
+  const size = e.object.size
+
+  const path = key.split("/")
+  const filename = path[path.length - 1]
+
+  return {
+    bucket,
+    path,
+    key,
+    filename,
+    size,
+  }
+}
+
+const isIngestConfig = function (config: any): config is IngestConfig {
+  return (
+    typeof config === "object" &&
```

Comment:
Possibly overkill for this project, however [io-ts](https://github.com/gcanti/io-ts) could make this a bit nicer.

Reply:
Thought about it with Jon but agreed that it would be overkill.